packages/destination-actions/src/destinations/gladly/gladly-operations.ts

```diff
@@ -0,0 +1,105 @@
+import { IntegrationError, RequestClient } from '@segment/actions-core'
+import { mappings } from './gladly-mappings'
+import type { Settings } from './generated-types'
+import type { GenericPayload, Customer } from './gladly-shared-types'
+
+export const API_VERSION = '/api/v1/'
+
+const resources = {
+  createItem: 'customers/{id}/conversation-items',
+  createCustomer: 'customer-profiles',
+  findCustomer: 'customer-profiles?{param}',
+  updateCustomer: 'customer-profiles/{id}'
+}
+
+export class Gladly {
+  url: string
+  request: RequestClient
+
+  constructor(settings: Settings, request: RequestClient) {
+    this.url = settings.url
+    this.request = request
+  }
+
+  createConversationItem = async (customerId: string, payload: GenericPayload) => {
+    this.validatePayload(payload)
+
+    const resource = resources['createItem'].replace('{id}', customerId)
+    const url = this.generateUrl(resource)
+    const json = mappings.conversationItem(payload)
+
+    const response = await this.request(url, { method: 'post', json })
+
+    if (!response || response.status !== 200) {
+      throw new IntegrationError('Unable to create customer conversation item')
+    }
+    return response
+  }
+
+  createCustomer = async (payload: GenericPayload) => {
+    this.validatePayload(payload)
+
+    const resource = resources['createCustomer']
+    const url = this.generateUrl(resource)
+    const json = mappings.createCustomer(payload)
+
+    const response = await this.request(url, { method: 'post', json })
+
+    if (!response || response.status !== 201) {
```

Comment:
from the documentation here: https://github.com/segmentio/action-destinations#http-request-options I think this code will never be executed as the error will be thrown. Or am I understanding this incorrectly?